### PR TITLE
Use derived constant in entity-react generator

### PR DESCRIPTION
**Pull request by Tcharl**
contributes to #14235

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by Tcharl** [comment]
Including correct bigdecimal support for react ;-).
I may create a ticket with the potential bugs I see while reading the code, I sometimes see some flags

**Comment by Tcharl** [comment]
@deepu105 a little review please if you can, mshima already spent too much hours on similar ones and I don't want to get him bored :-).

File: generators/entity-client/templates/common/src/test/javascript/cypress/integration/entity/entity.spec.ts.ejs [hunk 1]
```diff
@@ -173,14 +167,12 @@ describe('<%= entityClass %> e2e test', () => {
...
     const ownerSide = relationship.ownerSide;
-    const relationshipName = relationship.relationshipName;
-    const relationshipNameCapitalized = relationship.relationshipNameCapitalized;_%>
-    <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
+    const relationshipName = relationship.relationshipName; _%>
+    <%_ if (relationship.relationshipManyToOne || (relationship.relationshipOneToOne && ownerSide)) { _%>
     cy.setFieldSelectToLastOfEntity('<%= relationshipName %>');
 
-    <%_ } else if ((relationshipType === 'many-to-many' && ownerSide === true)) { _%>
+    <%_ } else if ((relationship.relationshipManyToMany && ownerSide)) { _%>
```
Comment (qmonmert) [inline generators/entity-client/templates/common/src/test/javascript/cypress/integration/entity/entity.spec.ts.ejs:175]:
``` diff
+ (relationship.relationshipManyToMany && ownerSide)
- ((relationship.relationshipManyToMany && ownerSide))
```
I think :)

File: generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs [hunk 2]
```diff
@@ -190,35 +178,33 @@ export default class <%= entityClass %>UpdatePage {
...
-        <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
+  <%_ } _%>
+<%_ }); _%>
+<%_ relationships.forEach((relationship) => {
+  const ownerSide = relationship.ownerSide;
+  const relationshipName = relationship.relationshipName; _%>
+  <%_ if (relationship.relationshipManyToOne || (relationship.relationshipOneToOne && ownerSide)) { _%>
         await this.<%= relationshipName %>SelectLastOption();
-        <%_ } else if ((relationshipType === 'many-to-many' && ownerSide === true)) { _%>
+  <%_ } else if ((relationship.relationshipManyToMany && ownerSide)) { _%>
```
Comment (qmonmert) [inline generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs:204]: same :)

File: jdl/jhipster/entity-options.js [hunk 3]
```diff
@@ -28,6 +28,7 @@ const ServiceTypes = {
 };
 
 const PaginationTypes = {
+  PAGER: 'pager',
```
Comment (mshima) [inline jdl/jhipster/entity-options.js:31]:
```suggestion
 ```

Pager doesn't exist anymore and should be removed.

File: utils/entity.js [hunk 4]
```diff
@@ -30,9 +30,9 @@ const { MapperTypes } = require('../jdl/jhipster/entity-options');
 const { OAUTH2 } = require('../jdl/jhipster/authentication-types');
 const { CommonDBTypes } = require('../jdl/jhipster/field-types');
 
-const { BOOLEAN } = CommonDBTypes;
+const { BOOLEAN, LONG } = CommonDBTypes;
 const { MAPSTRUCT } = MapperTypes;
-const { PAGINATION, INFINITE_SCROLL } = PaginationTypes;
+const { PAGER, PAGINATION, INFINITE_SCROLL } = PaginationTypes;
```
Comment (mshima) [inline utils/entity.js:35]:
```suggestion
const { PAGINATION, INFINITE_SCROLL } = PaginationTypes;
```

File: utils/entity.js [hunk 5]
```diff
@@ -92,6 +92,7 @@ function _derivedProperties(entityWithConfig) {
   const pagination = entityWithConfig.pagination;
   _.defaults(entityWithConfig, {
     paginationPagination: pagination === PAGINATION,
+    paginationPager: pagination === PAGER,
```
Comment (mshima) [inline utils/entity.js:95]:
```suggestion
```

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-reducer.spec.ts.ejs [hunk 6]
```diff
@@ -368,7 +368,7 @@ describe('Entities reducer tests', () => {
 
         }
       ];
-      await store.dispatch(partialUpdate({ id: <% if (primaryKey.type === 'Long') { %>1<% } else { %>'1'<% } %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
+      await store.dispatch(partialUpdate({ id: <% if (primaryKeyTypeLong) { %>1<% } else { %>'1'<% } %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-reducer.spec.ts.ejs:371]:
```suggestion
      await store.dispatch(partialUpdate({ id: <%= generateTestEntityId() %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
```

File: utils/entity.js [hunk 7]
```diff
@@ -221,6 +222,12 @@ function prepareEntityForTemplates(entityWithConfig, generator) {
   return entityWithConfig;
 }
 
+function _pkDerivedProperties(entityWithConfig) {
+  _.defaults(entityWithConfig, {
+    primaryKeyTypeLong: entityWithConfig.primaryKey.type === LONG,
+  });
+}
+
```
Comment (mshima) [inline utils/entity.js:230]:
```suggestion
```

File: utils/entity.js [hunk 8]
```diff
@@ -377,7 +384,7 @@ function prepareEntityPrimaryKeyForTemplates(entityWithConfig, generator, enable
       },
     };
   }
-
+  _pkDerivedProperties(entityWithConfig);
```
Comment (mshima) [inline utils/entity.js:387]:
```suggestion

```

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs [hunk 9]
```diff
@@ -33,13 +33,14 @@ import {
...
   byteSize,
   <%_ } _%>
-  Translate<% if (searchEngine !== false) { %>, translate, ICrudSearchAction<% } %>
+  Translate<% if (searchEngine) { %>, translate, ICrudSearchAction<% } %>
   <% if (fieldsContainDate) { %>, TextFormat<% } %>
-  <%_ if (pagination !== 'no') { _%>
+  <%_ if (!paginationNo) { _%>
   , getSortState, IPaginationBaseState
-  <%_ if (pagination === 'pagination' || pagination === 'pager') { _%>
+  <%_ if (paginationPagination || paginationPager) { _%>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity.tsx.ejs:40]:
```suggestion
  <%_ if (paginationPagination) { _%>
```

File: generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs [hunk 10]
```diff
@@ -33,65 +33,56 @@ if (enableTranslation) {
...
-        const ownerSide = relationship.ownerSide;
-        const relationshipName = relationship.relationshipName;
-        const relationshipFieldName = relationship.relationshipFieldName; _%>
-    <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'many-to-many' && ownerSide === true) || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
+  <%_ } _%>
+<%_ }); _%>
+<%_ relationships.forEach((relationship) => {
+    const ownerSide = relationship.ownerSide;
+    const relationshipName = relationship.relationshipName; _%>
+  <%_ if (relationship.relationshipManyToOne || (relationship.relationshipManyToMany && ownerSide) || (relationship.relationshipOneToOne && ownerSide)) { _%>
```
Comment (mshima) [inline generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs:67]:
```suggestion
  <%_ if (ownerSide) { _%>
```

relationshipManyToOne is implicit ownerSide and we started setting it at jhipster 7.

**Comment by mshima** [comment]
There are changes in the generated react code, I will let someone else say if the change is correct.

File: jdl/jhipster/entity-options.js (hunk 3 above)
Comment (Tcharl) [reply jdl/jhipster/entity-options.js:31]: It was always present in some templates: I just changed equality by variables :-)

File: generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs (hunk 10 above)
Comment (Tcharl) [reply generators/entity-client/templates/react/src/test/javascript/e2e/entities/entity-update-page-object.ts.ejs:67]: What about oneToMany?

**Comment by Tcharl** [comment]
The changes are on the bigdecimal handling: it looks like it was not supported in react before this update

File: generators/entity-client/templates/react/src/main/webapp/app/entities/entity-reducer.spec.ts.ejs [hunk 11]
```diff
@@ -368,7 +368,7 @@ describe('Entities reducer tests', () => {
 
         }
       ];
-      await store.dispatch(partialUpdate({ id: <% if (primaryKey.type === 'Long') { %>1<% } else { %>'1'<% } %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
+      await store.dispatch(partialUpdate({ id: <%= generateTestEntityId(primaryKey.type) %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
```
Comment (mshima) [inline generators/entity-client/templates/react/src/main/webapp/app/entities/entity-reducer.spec.ts.ejs:371]:
```suggestion
      await store.dispatch(partialUpdate({ id: <%- generateTestEntityId(primaryKey.type) %> })).then(() => expect(store.getActions()).toEqual(expectedActions));
```

**Pull request merged**